File: Code/SimpleBudget.API/Models/Reports/MonthlyModel.cs
namespace SimpleBudget.API.Models
{
    public class MonthlyModel
    {
        public int SelectedYear { get; set; }
        public int SelectedMonth { get; set; }
        public bool ShowWeekly { get; set; }

        public List<MonthlyWalletModel> Wallets { get; set; } = default!;
        public List<MonthlyCategoryModel> Categories { get; set; } = default!;
        public List<MonthlySummaryModel> Summaries { get; set; } = default!;
        public List<int> Years { get; set; } = default!;
        public List<string> MonthNames { get; set; } = default!;

        public string ValueFormatCAD { get; set; } = default!;
    }
}

File: Code/client/src/api/models/MonthlyModel.ts
import { MonthlyCategoryModel } from "./MonthlyCategoryModel";
import { MonthlySummaryModel } from "./MonthlySummaryModel";
import { MonthlyWalletModel } from "./MonthlyWalletModel";

export interface MonthlyModel
{
    selectedYear: number;
    selectedMonth: number;
    showWeekly: boolean;
    wallets: MonthlyWalletModel[];
    categories: MonthlyCategoryModel[];
    summaries: MonthlySummaryModel[];
    years: number[];
    valueFormatCAD: string;
}
